// @clerk/react/dist/internal.mjs
import {
  ClerkProvider,
  IS_REACT_SHARED_VARIANT_COMPATIBLE,
  MultisessionAppSupport
} from "./chunk-4FWGU4X2.mjs";
import {
  incompatibleRoutingWithPathProvidedError,
  noPathProvidedError,
  useDerivedAuth
} from "./chunk-XE3Y43DU.mjs";
import {
  errorThrower,
  setErrorThrowerOptions
} from "./chunk-RQWALB2R.mjs";
import "./chunk-E5QRIS4Z.mjs";

// src/hooks/useRoutingProps.ts
function useRoutingProps(componentName, props, routingOptions) {
  const path = props.path || (routingOptions == null ? void 0 : routingOptions.path);
  const routing = props.routing || (routingOptions == null ? void 0 : routingOptions.routing) || "path";
  if (routing === "path") {
    if (!path) {
      return errorThrower.throw(noPathProvidedError(componentName));
    }
    return {
      ...routingOptions,
      ...props,
      routing: "path"
    };
  }
  if (props.path) {
    return errorThrower.throw(incompatibleRoutingWithPathProvidedError(componentName));
  }
  return {
    ...routingOptions,
    ...props,
    path: void 0
  };
}

// src/internal.ts
import {
  clerkJSScriptUrl,
  buildClerkJSScriptAttributes,
  clerkUIScriptUrl,
  buildClerkUIScriptAttributes,
  setClerkJSLoadingErrorPackageName,
  clerkJsScriptUrl,
  buildClerkJsScriptAttributes,
  setClerkJsLoadingErrorPackageName
} from "@clerk/shared/loadClerkJsScript";
var InternalClerkProvider = ClerkProvider;
export {
  IS_REACT_SHARED_VARIANT_COMPATIBLE,
  InternalClerkProvider,
  MultisessionAppSupport,
  buildClerkJSScriptAttributes,
  buildClerkJsScriptAttributes,
  buildClerkUIScriptAttributes,
  clerkJSScriptUrl,
  clerkJsScriptUrl,
  clerkUIScriptUrl,
  setClerkJSLoadingErrorPackageName,
  setClerkJsLoadingErrorPackageName,
  setErrorThrowerOptions,
  useDerivedAuth,
  useRoutingProps
};
